mantle/platform/qemu.go

```diff
@@ -243,16 +254,32 @@ func virtio(device, args string) string {
 	return fmt.Sprintf("virtio-%s-%s,%s", device, suffix, args)
 }
 
-// addQcow2Disk adds a disk image from a file descriptor,
+// addDisk adds a disk image from a file descriptor,
 // mounted read-write, formatted qcow2.
-func (builder *QemuBuilder) addQcow2DiskFd(fd *os.File, channel string, options []string) {
+func (builder *QemuBuilder) addDiskFd(fd *os.File, channel string, disk *Disk, options []string) {
 	opts := ""
 	if len(options) > 0 {
 		opts = "," + strings.Join(options, ",")
 	}
 	fdset := builder.AddFd(fd)
-	id := fmt.Sprintf("d%d", builder.diskId)
 	builder.diskId += 1
+	id := fmt.Sprintf("d%d", builder.diskId)
+
+	if disk.MultiPathDisk {
+		builder.Append("-device", "virtio-scsi-pci,id=scsi")
```

Comment:
If we try to add multiple multipath disks - will this fail since we're appending multiple with the same `id=scsi`?  I think we either need to only append this once, or create a separate bus per disk?

Reply:
Ah, that is true. Testing a fix now.